### PR TITLE
Zilean scraper

### PR DESCRIPTION
**Pull request by mhdzumair**
## Summary by CodeRabbit

- **New Features**
  - Introduced the ability to scrape and display streams from Zilean DMM for both movies and series.
  - Added a feature toggle for scraping from Zilean in user settings.

- **Enhancements**
  - Updated user settings to include Zilean DMM streams in selected catalogs.
  - Improved handling of cached torrents and magnet links in the user experience.

### CODE REVIEW

File: scrapers/zilean.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        return response.json()
+
+
+async def scrap_movie_streams_from_zilean(
+    video_id: str, title: str,
+) -> list[TorrentStreams]:
+    """
+    Get streams by text from zilean DMM.
+    """
+    url = f"{settings.zilean_url}/dmm/filtered?Query={title}"
```
Comment (mhdzumair) [inline scrapers/zilean.py:77]: zilean using parsett library to parse the data. I'm also going to use the same. However that implementation is bit old and im adding new parsing capability. 

zilean also support raw search without filtering with matching queries. 

Ex: 
```
curl --location 'http://zilean.zilean:8181/dmm/search' \
--header 'Content-Type: application/json' \
--data '{"queryText": "Hit Man"}'
```

Whats your thought? do we use already parsed data or get the data and parse it from our side?

Comment (ragmehos) [reply scrapers/zilean.py:77]: We can parse it from the raw title if you want and then use the parsett library to parse the data. We can use the same endpoint I am using here for now and that has the raw title anyways, and you can add the parsett parsing once this is checked in and replace how you get the info?

So anyways this endpoint provides more info and can be used to enhance whatever we need?

File: scrapers/zilean.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    for stream in stream_data:
+        torrent_stream = await TorrentStreams.get(stream["infoHash"])
+        if torrent_stream:
+            continue
+
+        if is_contain_18_plus_keywords(stream["rawTitle"]):
+            logging.warning(f"Stream contains 18+ keywords: {stream['rawTitle']}")
+            continue
+
+        metadata = PTN.parse(stream['rawTitle'])
```
[the remark this answers is not in the document]
Comment (ragmehos) [reply scrapers/zilean.py:126]: Done

File: scrapers/zilean.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+        # Create new stream
+        torrent_stream = TorrentStreams(
+            id=stream["infoHash"],
+            torrent_name=stream["rawTitle"],
+            announce_list=[],
+            size=convert_size_to_bytes(extract_size_string(stream["size"])),
```
[the remark this answers is not in the document]
Comment (ragmehos) [reply scrapers/zilean.py:183]: Done

**Comment by mhdzumair** [comment]
Merging this and ill work on improvments

**Pull request merged**